src/analysis/processing/qgsalgorithmcellstatistics.cpp

```diff
@@ -0,0 +1,337 @@
+                      "   <li>Minimum</li>"
+                      "   <li>Maximum</li>"
+                      "   <li>Minority (most frequent value)</li>"
+                      "   <li>Majority (least frequent value)</li>"
+                      "   <li>Range (max-min)</li>"
+                      "   <li>Variety (count of unique values)</li>"
+                      "</ul> "
+                      "Input raster layers that do not match the cell size of the reference raster layer will be "
+                      "resampled using nearest neighbor resampling. The output raster data type will be set to "
+                      "the most complex data type present in the input datasets except when using the functions "
+                      "Mean and Standard deviation (data type is always Float32) or Count and Variety (data type is always Int32).\n"
+                      "<i>Calculation details - general:</i> NoData values in any of the input layers will result in a NoData cell output if the Ignore NoData parameter is not set.\n"
+                      "<i>Calculation details - Count:</i> Count will always result in the number of cells without NoData values at the current cell location.\n"
+                      "<i>Calculation details - Median:</i> If the number of input layers is even, the median will be calculated as the "
+                      "arithmetic mean of the two middle values of the ordered cell input values.\n"
+                      "<i>Calculation details - Minority/Majority:</i> If no unique minority or majority could be found, the result is NoData, except all "
+                      "input cell values are equal." );
+}
+
+QgsCellStatisticsAlgorithm *QgsCellStatisticsAlgorithm::createInstance() const
+{
+  return new QgsCellStatisticsAlgorithm();
+}
+
+void QgsCellStatisticsAlgorithm::initAlgorithm( const QVariantMap & )
+{
+  addParameter( new QgsProcessingParameterMultipleLayers( QStringLiteral( "INPUTS" ),
+                QObject::tr( "Input layers" ), QgsProcessing::TypeRaster ) );
+
+  QStringList statistics = QStringList();
+  statistics << QObject::tr( "Sum" )
+             << QObject::tr( "Count" )
+             << QObject::tr( "Mean" )
+             << QObject::tr( "Median" )
+             << QObject::tr( "Standard deviation" )
+             << QObject::tr( "Variance" )
+             << QObject::tr( "Minimum" )
+             << QObject::tr( "Maximum" )
+             << QObject::tr( "Minority" )
+             << QObject::tr( "Majority" )
+             << QObject::tr( "Range" )
+             << QObject::tr( "Variety" );
+
+  addParameter( new QgsProcessingParameterEnum( QStringLiteral( "STATISTIC" ), QObject::tr( "Statistic" ),  statistics, false, 0, false ) );
+  addParameter( new QgsProcessingParameterBoolean( QStringLiteral( "IGNORE_NODATA" ), QObject::tr( "Ignore NoData values" ), true ) );
+
+  addParameter( new QgsProcessingParameterRasterLayer( QStringLiteral( "REF_LAYER" ), QObject::tr( "Reference layer" ) ) );
```

Comment:
```suggestion
  addParameter( new QgsProcessingParameterRasterLayer( QStringLiteral( "REFERENCE_LAYER" ), QObject::tr( "Reference layer" ) ) );
```

And let's avoid the abbreviation here -- it makes the api trickier for non english native speakers to understand